fendermint/vm/interpreter/src/fvm/checkpoint.rs

```diff
@@ -322,23 +297,47 @@ fn should_create_checkpoint<DB>(
     gateway: &GatewayCaller<DB>,
     state: &mut FvmExecState<DB>,
     height: Height,
-) -> anyhow::Result<Option<checkpoint::SubnetID>>
+) -> anyhow::Result<Option<(Vec<checkpoint::IpcEnvelope>, checkpoint::SubnetID)>>
 where
     DB: Blockstore,
 {
-    if gateway.enabled(state)? {
-        let id = gateway.subnet_id(state)?;
-        let is_root = id.route.is_empty();
-
-        if !is_root && height.value() % gateway.bottom_up_check_period(state)? == 0 {
-            let id = checkpoint::SubnetID {
-                root: id.root,
-                route: id.route,
-            };
-            return Ok(Some(id));
-        }
+    if !gateway.enabled(state)? {
+        return Ok(None);
+    }
+
+    let id = gateway.subnet_id(state)?;
+    let is_root = id.route.is_empty();
+
+    if is_root {
+        return Ok(None);
+    }
+
+    let batch = gateway.bottom_up_msg_batch(state, height.into())?;
+
+    if batch.block_height.as_u64() != 0 {
+        tracing::debug!(
+            height = height.value(),
+            "bottom up msg batch exists at height"
+        );
+    } else if height.value() % gateway.bottom_up_check_period(state)? == 0 {
+        tracing::debug!(
+            height = height.value(),
+            "bottom up checkpoint period reached height"
+        );
+    } else {
+        tracing::debug!(
+            height = height.value(),
+            "height has no message batch nor reached checkpoint period "
+        );
```

Comment:
This will be a bit spammy on debug level at every block height.

Reply:
yeah, probably the batch full log message can stay, this one can go